lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/base-edit-task-filter-cloud.component.ts

```diff
@@ -145,8 +147,12 @@ export abstract class BaseEditTaskFilterCloudComponent<T> implements OnInit, OnC
     }
 
     ngOnChanges(changes: SimpleChanges) {
-        const id = changes['id'];
-        if (id && id.currentValue !== id.previousValue) {
+        const { id, taskFilter } = changes;
+        if(taskFilter && taskFilter.currentValue?.processInstanceId !== taskFilter.previousValue?.processInstanceId){
+            this.taskFilterProperties = this.createAndFilterProperties();
+            this.taskFilterActions = this.createAndFilterActions();
+            this.buildForm(this.taskFilterProperties);
+        } else if (id && id.currentValue !== id.previousValue) {
```

Comment:
Are you sure this is not breaking anything? Before the flow was

if (id && id.currentValue !== id.previousValue) {
            this.retrieveTaskFilterAndBuildForm();
}

and now the same condition is within an else if. It changes the flow and for that condition to be there I suppose there was a reason.

Have u tested a scenario were process instance id changes but also the filter id? Is there such scenario where the filter id would change but the else if condition (line 55) not reached because it is inside an else if instead of a plain if that was before?